### ISSUE TITLE
buildSearchableOptions takes very long

### ISSUE THREAD

**Issue by matkoch** [JetBrains/gradle-intellij-plugin#617]
It takes about 1:20 to initiate a debug session with Rider plugins. About 1 minute is taken in the `buildSearchableOptions`. It doesn't seem to be related to #616, as I don't see any download notification:

```
2021-03-22 01:23:31,963 [  17428]   WARN - on.HighlighterRegistrationHost - ColorsScheme 'ReSharperLight' is not found in IDEA, omitting scheme patching 
2021-03-22 01:23:31,964 [  17429]   WARN - on.HighlighterRegistrationHost - ColorsScheme 'ReSharperDark' is not found in IDEA, omitting scheme patching 
2021-03-22 01:23:31,972 [  17437]   WARN - on.HighlighterRegistrationHost - ColorsScheme 'VisualStudioLight' is not found in IDEA, omitting scheme patching 
2021-03-22 01:23:31,972 [  17437]   WARN - on.HighlighterRegistrationHost - ColorsScheme 'VisualStudioDark' is not found in IDEA, omitting scheme patching 
2021-03-22 01:23:33,774 [  19239]   WARN - nSystem.impl.ActionManagerImpl - keymap "Eclipse" not found [Plugin: com.intellij] 
2021-03-22 01:23:33,774 [  19239]   WARN - nSystem.impl.ActionManagerImpl - keymap "NetBeans 6.5" not found [Plugin: com.intellij] 
2021-03-22 01:23:33,779 [  19244]   WARN - nSystem.impl.ActionManagerImpl - keymap "Eclipse (Mac OS X)" not found [Plugin: com.intellij] 
2021-03-22 01:23:33,790 [  19255]   WARN - nSystem.impl.ActionManagerImpl - keymap "QtCreator (Mac OS X)" not found [Plugin: com.intellij] 
2021-03-22 01:23:33,790 [  19255]   WARN - nSystem.impl.ActionManagerImpl - keymap "QtCreator" not found [Plugin: com.intellij] 
2021-03-22 01:23:33,792 [  19257]   WARN - nSystem.impl.ActionManagerImpl - keymap "QtCreator (Mac OS X)" not found [Plugin: com.intellij] 
2021-03-22 01:23:33,792 [  19257]   WARN - nSystem.impl.ActionManagerImpl - keymap "QtCreator" not found [Plugin: com.intellij] 
2021-03-22 01:23:33,884 [  19349]   WARN - nSystem.impl.ActionManagerImpl - keymap "Eclipse" not found [Plugin: org.jetbrains.plugins.github] 
2021-03-22 01:23:33,884 [  19349]   WARN - nSystem.impl.ActionManagerImpl - keymap "Eclipse (Mac OS X)" not found [Plugin: org.jetbrains.plugins.github] 
2021-03-22 01:23:33,884 [  19349]   WARN - nSystem.impl.ActionManagerImpl - keymap "Eclipse (Mac OS X)" not found [Plugin: org.jetbrains.plugins.github] 
2021-03-22 01:23:33,900 [  19365]   WARN - nSystem.impl.ActionManagerImpl - keymap "Eclipse" not found [Plugin: JavaScript] 
2021-03-22 01:23:48,199 [  33664]   WARN - #com.intellij.ui.jcef.JBCefApp - JCEF is manually disabled in headless env via 'ide.browser.jcef.headless.enabled=false' 
2021-03-22 01:24:00,341 [  45806]   WARN - m.plugin.external.tools.TfTool - No tf command was detected 
```

**Comment by zolotov** [comment]
please provide CPU snapshot of the process that spawned by `buildSearchableOptions` task

**Comment by zolotov** [comment]
if you use IntelliJ you can attach profiler to process like that:

![image](https://user-images.githubusercontent.com/140920/113193353-2c23ee00-9260-11eb-98cf-30425b7c3ada.png)

**Comment by matkoch** [comment]
i have a short question before I further investigate this:

I recently created new run configs for our Rider plugin template. These are chained run configs, where one invoked `buildPlugin` and the other did `runIde`. Now I changed `buildPlugin` to `compileKotlin` and everything seems to work fast. How exactly are the dependencies defined? Is `buildSearchableOptions` not part of `runIde` ? (Sorry couldn't find an answer myself)
